scripts/publisher_example.py

```diff
@@ -0,0 +1,72 @@
+import time
+import rclpy
+from rclpy.node import Node
+import threading
+
+from std_msgs.msg import String
+
+
+class SelfSpinningNode(Node):
+    def __init__(self, name):
+        super().__init__(name)
+        self.executor_ = rclpy.executors.SingleThreadedExecutor()
+        self.executor_.add_node(self)
+        self.executor_thread_ = threading.Thread(target=self.executor_.spin, daemon=True)
+
+    def start_spinning(self):
+        self.executor_thread_.start()
+
+class MinimalTalker(SelfSpinningNode):
+
+    def __init__(self):
+        super().__init__('minimal_talker')
+        self.i = 0
+        self.timer = None
+        self.publisher_ = None
+        self.subscriber_ = None
+
+    def run_timer(self):
+        timer_period = 0.5  # seconds
```

Comment:
usually it's better to name variables rather than commenting. For instance `time_period_seconds`